modules/api/src/main/java/net/countercraft/movecraft/MovecraftRepair.java

```diff
@@ -0,0 +1,30 @@
+package net.countercraft.movecraft;
+
+
+import com.sk89q.worldedit.extent.clipboard.Clipboard;
+import com.sk89q.worldguard.protection.regions.ProtectedRegion;
+import net.countercraft.movecraft.craft.Craft;
+import net.countercraft.movecraft.utils.Pair;
+import org.bukkit.Material;
+import org.bukkit.World;
+import org.bukkit.block.Sign;
+import org.bukkit.util.Vector;
+
+import java.util.ArrayDeque;
+import java.util.HashMap;
+
+
+public abstract class MovecraftRepair {
+    public abstract boolean saveCraftRepairState(Craft craft, Sign sign, String repairStateName);
+    public abstract boolean saveRegionRepairState(World world, ProtectedRegion region);
+    public abstract Clipboard loadCraftRepairStateClipboard(Craft craft, Sign sign, String repairStateName, World bukkitWorld);
+    public abstract Clipboard loadRegionRepairStateClipboard(String repairStateName, World bukkitWorld);
+    public abstract HashMap<Material, Double> getMissingBlocks(String repairName);
+    public abstract ArrayDeque<Pair<Vector, Vector>> getMissingBlockLocations(String repairName);
+    public abstract long getNumDiffBlocks(String repairName);
+    public abstract org.bukkit.util.Vector getDistanceFromSignToLowestPoint(Clipboard clipboard);
+    public abstract org.bukkit.util.Vector getDistance(String repairName);
```

Comment:
This does not need to exist at the API level

Reply:
For clarification, there's only one implementation of this, and it exists in its own module - there is no need for an abstract class to define it.